pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java

```diff
@@ -63,47 +64,67 @@ int getInitialRowsThreshold() {
         long targetSegmentNumRows = (long) (IDEAL_SEGMENT_SIZE_BYTES * _latestSegmentRowsToSizeRatio);
         targetSegmentNumRows = capNumRowsIfOverflow(targetSegmentNumRows);
+        LOGGER.info(
+            "Committing segment zk metadata is not available, using prev ratio {}, setting rows threshold for {} as {}",
+            _latestSegmentRowsToSizeRatio, newSegmentName, targetSegmentNumRows);
         newSegmentZKMetadata.setSizeThresholdToFlushSegment((int) targetSegmentNumRows);
       } else {
-        LOGGER.info("Committing segment zk metadata is not available, setting default rows threshold for segment {}",
-            newSegmentZKMetadata.getSegmentName());
+        LOGGER.info("Committing segment zk metadata is not available, setting threshold for {} as {}",
+            newSegmentName, INITIAL_ROWS_THRESHOLD);
         newSegmentZKMetadata.setSizeThresholdToFlushSegment(INITIAL_ROWS_THRESHOLD);
       }
       return;
     }
 
     long committingSegmentSizeBytes = committingSegmentDescriptor.getSegmentSizeBytes();
     if (committingSegmentSizeBytes <= 0) { // repair segment case
+      final int targetNumRows = committingSegmentZKMetadata.getSizeThresholdToFlushSegment();
       LOGGER.info(
-          "Committing segment size is not available, setting thresholds for segment {} from previous segment {}",
-          newSegmentZKMetadata.getSegmentName(), committingSegmentZKMetadata.getSegmentName());
-      newSegmentZKMetadata.setSizeThresholdToFlushSegment(committingSegmentZKMetadata.getSizeThresholdToFlushSegment());
+          "Committing segment size is not available, setting thresholds from previous segment for {} as {}",
+          newSegmentZKMetadata.getSegmentName(), targetNumRows);
+      newSegmentZKMetadata.setSizeThresholdToFlushSegment(targetNumRows);
       return;
     }
 
     long timeConsumed = System.currentTimeMillis() - committingSegmentZKMetadata.getCreationTime();
     long numRowsConsumed = committingSegmentZKMetadata.getTotalRawDocs();
     int numRowsThreshold = committingSegmentZKMetadata.getSizeThresholdToFlushSegment();
-    LOGGER.info("Time consumed:{}  Num rows consumed:{} Num rows threshold:{} Committing segment size bytes:{}",
-        TimeUtils.convertMillisToPeriod(timeConsumed), numRowsConsumed, numRowsThreshold, committingSegmentSizeBytes);
+    LOGGER.info("{}: Data from committing segment: Time {}  numRows {} threshold {} segmentSize(bytes) {}",
+        newSegmentName, TimeUtils.convertMillisToPeriod(timeConsumed), numRowsConsumed, numRowsThreshold,
+        committingSegmentSizeBytes);
 
     double currentRatio = (double) numRowsConsumed / committingSegmentSizeBytes;
-    if (new LLCSegmentName(committingSegmentZKMetadata.getSegmentName()).getPartitionId() == 0) {
+    // Compute segment size to rows ratio only from partition 0.
+    // If we consider all partitions then it is likely that we will assign a much higher weight to the most
```

Comment:
Would taking average of all partitions avoid the skewed weight issue?

Reply:
No, but why complicate the logic taking averages of all partitions?